### ISSUE TITLE
Arrays.DisallowPartiallyKeyed.DisallowedPartiallyKeyed false positive when array with closure with foreach

### ISSUE THREAD

**Issue by Levivb** [slevomat/coding-standard#1571]
I found a false positive in `SlevomatCodingStandard.Arrays.DisallowPartiallyKeyed.DisallowedPartiallyKeyed` when an indexed array contains a closure containing a foreach with `$key => $value` syntax.

Snippet
```
<?php

[
    'question' => [
        'bail',
        'array',
        'required',
        static function (array $value): array {
            foreach ($value as $x => $z) {
                $x + $z; // uselessly using the variables to prevent `Unused variable $x.` sniff errors
            }
        },
    ],
];
```

Rule: `SlevomatCodingStandard.Arrays.DisallowPartiallyKeyed.DisallowedPartiallyKeyed`

Gives output:
```
$ composer style -- app/Console/test.php -s
> vendor/bin/phpcs --standard=phpcs.xml 'app/Console/test.php' '-s'
E 1 / 1 (100%)



FILE: app/Console/test.php
-------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------
 6 | ERROR | Partially keyed array disallowed.
   |       | (SlevomatCodingStandard.Arrays.DisallowPartiallyKeyed.DisallowedPartiallyKeyed)
-------------------------------------------------------------------------------------------------------------------------------

Time: 101ms; Memory: 38.94MB

Script vendor/bin/phpcs --standard=phpcs.xml handling the style event returned with error code 1
```

It seems the `$x => $z` syntax messes up this sniff

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/a32f46c71dff1d20cf33b7f83690e478ffec42d4

**Comment by Levivb** [comment]
Hero!
